ote/src/clj/ote/tasks/gtfs.clj

```diff
@@ -32,34 +33,55 @@
     "GTFS" :gtfs
     "Kalkati.net" :kalkati))
 
-(defn fetch-and-mark-gtfs-interface! [config db]
+(defn mark-gtfs-package-imported! [db gtfs-data]
+  (specql/update! db ::t-service/external-interface-description
+                  {::t-service/gtfs-imported (java.sql.Timestamp. (System/currentTimeMillis))}
+                  {::t-service/id (:id gtfs-data)})) ;; external-interface-description.id, not service id.
+
+(defn fetch-next-gtfs-interface! [config db]
   (tx/with-transaction
     db
     (let [blacklisted-operators {:blacklist (if (empty? (:no-gtfs-update-for-operators config))
                                               #{-1}         ;; this is needed for postgres NOT IN conditional
                                               (:no-gtfs-update-for-operators config))}
           gtfs-data (first (select-gtfs-urls-update db blacklisted-operators))]
       (when gtfs-data
-        (specql/update! db ::t-service/external-interface-description
-                        {::t-service/gtfs-imported (java.sql.Timestamp. (System/currentTimeMillis))}
-                        {::t-service/id (:id gtfs-data)}))
+       (mark-gtfs-package-imported! db gtfs-data))
       gtfs-data)))
 
-(defn update-one-gtfs! [config db upload-s3?]
+(defn fetch-given-gtfs-interface!
+  "Get gtfs package data from database for given service."
+  [db service-id]
+  (let [gtfs-data (first (select-gtfs-url-for-service db {:service-id service-id}))]
+    (when gtfs-data
+      (mark-gtfs-package-imported! db gtfs-data))
+    gtfs-data))
+
+(defn update-one-gtfs!
+  ([config db upload-s3?]
+   (update-one-gtfs! config db upload-s3? nil))
+  ([config db upload-s3? service-id]
   ;; Ensure that gtfs-import flag is enabled
   ;; upload-s3? should be false when using local environment
   (let [{:keys [id url operator-id ts-id last-import-date format license] :as gtfs-data}
-        (fetch-and-mark-gtfs-interface! config db)]
+        ;; Load next gtfs package or package that is related to given service-id
+        (if (nil? service-id)
+          (fetch-next-gtfs-interface! config db)
+          (fetch-given-gtfs-interface! db service-id))
+        force-download? (integer? service-id)]
     (if (nil? gtfs-data)
-      (log/debug "No gtfs files to upload.")
+      (do
+        (log/debug "No gtfs files to upload.")
+        (http/transit-response "No gtfs files to upload." 400))
```

Comment:
A 2xx status code should be used when an operation completed without errors.
In this case 200 or if we want to be exact, 204 No Content.

400 is not the best, because 
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
> 400 Bad Request
> The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.

Reply:
I added a notification box to inform users.